coordinator/gscoordinator/utils.py

```diff
@@ -830,67 +830,141 @@ def _pre_process_for_output_graph_op(op, op_result_pool, key_to_op, **kwargs):
     )
 
 
-def _pre_process_for_project_to_simple_op(op, op_result_pool, key_to_op, **kwargs):
+def _pre_process_for_project_to_simple_op(  # noqa: C901
+    op, op_result_pool, key_to_op, **kwargs
+):
     # for nx graph
     if op.attr[types_pb2.GRAPH_TYPE].graph_type in (
         graph_def_pb2.DYNAMIC_PROJECTED,
         graph_def_pb2.ARROW_FLATTENED,
     ):
         return
-    assert len(op.parents) == 1
+
+    def __check_v_prop_exists_in_all_v_labels(schema, prop):
+        exists = True
+        for v_label in schema.vertex_labels:
+            exists = exists and schema.vertex_property_exists(v_label, prop)
+        return exists
+
+    def __check_e_prop_exists_in_all_e_labels(schema, prop):
+        exists = True
+        for e_label in schema.edge_labels:
+            exists = exists and schema.edge_property_exists(e_label, prop)
+        return exists
+
     # get parent graph schema
+    assert len(op.parents) == 1
     key_of_parent_op = op.parents[0]
     r = op_result_pool[key_of_parent_op]
     schema = GraphSchema()
     schema.from_graph_def(r.graph_def)
     graph_name = r.graph_def.key
-    check_argument(
-        schema.vertex_label_num == 1,
-        "Cannot project to simple, vertex label number is not one.",
-    )
-    check_argument(
-        schema.edge_label_num == 1,
-        "Cannot project to simple, edge label number is not one.",
-    )
-    v_label = schema.vertex_labels[0]
-    e_label = schema.edge_labels[0]
-    relation = (v_label, v_label)
-    check_argument(
-        relation in schema.get_relationships(e_label),
-        f"Cannot project to simple, Graph doesn't contain such relationship: {v_label} -> {e_label} <- {v_label}.",
-    )
-    v_props = schema.get_vertex_properties(v_label)
-    e_props = schema.get_edge_properties(e_label)
-    check_argument(len(v_props) <= 1)
-    check_argument(len(e_props) <= 1)
-    v_label_id = schema.get_vertex_label_id(v_label)
-    e_label_id = schema.get_edge_label_id(e_label)
-    v_prop_id, vdata_type = (v_props[0].id, v_props[0].type) if v_props else (-1, None)
-    e_prop_id, edata_type = (e_props[0].id, e_props[0].type) if e_props else (-1, None)
-    oid_type = schema.oid_type
-    vid_type = schema.vid_type
+
+    if schema.vertex_label_num == 0:
+        raise RuntimeError("Project simple graph failed due to vertex label num is 0.")
+    if schema.edge_label_num == 0:
+        raise RuntimeError("Project simple graph failed due to edge label num is 0.")
```

Comment:
`raise RuntimeError("Failed to project to simple graph failed as no edge exists in this graph.")`